cmd/lncli/cmd_pay.go

```diff
@@ -0,0 +1,581 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/lightninglabs/protobuf-hex-display/jsonpb"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/record"
+	"github.com/lightningnetwork/lnd/routing/route"
+	"github.com/urfave/cli"
+)
+
+var (
+	cltvLimitFlag = cli.UintFlag{
+		Name: "cltv_limit",
+		Usage: "the maximum time lock that may be used for " +
+			"this payment",
+	}
+
+	lastHopFlag = cli.StringFlag{
+		Name: "last_hop",
+		Usage: "pubkey of the last hop (penultimate node in the path) " +
+			"to route through for this payment",
+	}
+
+	dataFlag = cli.StringFlag{
+		Name: "data",
+		Usage: "attach custom data to the payment. The required " +
+			"format is: <record_id>=<hex_value>,<record_id>=" +
+			"<hex_value>,.. For example: --data 3438382=0a21ff. " +
+			"Custom record ids start from 65536.",
+	}
+)
+
+// paymentFlags returns common flags for sendpayment and payinvoice.
+func paymentFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:  "pay_req",
+			Usage: "a zpay32 encoded payment request to fulfill",
+		},
+		cli.Int64Flag{
+			Name: "fee_limit",
```

Comment:
not blocking, but could be a good time to rename this to `fee_limit_sat` to match `fee_limit_percent` so they both have units

Reply:
I am in doubt about it. It can be done at any time really too. This parameter already existed.